android/src/main/java/com/onegini/mobile/sdk/reactnative/RNOneginiSdk.kt

```diff
@@ -369,50 +367,25 @@ class RNOneginiSdk(reactContext: ReactApplicationContext) : ReactContextBaseJava
-                handleSubmitPinActionProvide(pinFlow, pin, promise)
-                return
-            }
-            Constants.PIN_ACTION_CANCEL -> {
-                handleSubmitPinActionCancel(pinFlow, promise)
-                return
-            }
-            else -> {
-                promise.reject(OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.code, OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.message + ". Incorrect action supplied: $action")
-                return
-            }
-        }
-    }
-
-    private fun handleSubmitPinActionProvide(pinFlow: String?, pin: String, promise: Promise) {
-        when (pinFlow) {
-            PinFlow.Authentication.toString() -> {
-                oneginiSDK.pinAuthenticationRequestHandler.acceptAuthenticationRequest(pin.toCharArray())
-                return promise.resolve(null)
-            }
-            PinFlow.Create.toString() -> {
-                oneginiSDK.createPinRequestHandler.onPinProvided(pin.toCharArray())
-                return promise.resolve(null)
-            }
-        }
-    }
-
-    private fun handleSubmitPinActionCancel(pinFlow: String?, promise: Promise) {
-        when (pinFlow) {
-            PinFlow.Authentication.toString() -> {
-                oneginiSDK.pinAuthenticationRequestHandler.denyAuthenticationRequest()
-                return promise.resolve(null)
-            }
-            PinFlow.Create.toString() -> {
-                return try {
-                    oneginiSDK.createPinRequestHandler.cancelPin()
-                    promise.resolve(null)
-                } catch (exception: OneginiReactNativeException) {
-                    promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
+    fun submitPin(pinFlow: String?, pin: String?, promise: Promise) {
+        when (pin) {
+            null -> promise.rejectWithNullError(Pin.paramName, Pin.type)
+            else -> when (pinFlow) {
+                PinFlow.Authentication.toString() -> {
+                    return try {
+                        oneginiSDK.pinAuthenticationRequestHandler.acceptAuthenticationRequest(pin.toCharArray())
+                        promise.resolve(null)
+                    } catch (exception: OneginiReactNativeException) {
+                        promise.reject(OneginiWrapperErrors.AUTHENTICATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.AUTHENTICATION_NOT_IN_PROGRESS.message)
+                    }
+                }
+                PinFlow.Create.toString() -> {
+                    return try {
+                        oneginiSDK.createPinRequestHandler.onPinProvided(pin.toCharArray())
+                        promise.resolve(null)
+                    } catch (exception: OneginiReactNativeException) {
+                        promise.reject(OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.code, OneginiWrapperErrors.REGISTRATION_NOT_IN_PROGRESS.message)
+                    }
```

Comment:
There quite a lot of nesting in that method. I would move that nested `when` to separate method to improve readability